### DISCUSSION TITLE
WebRTC without WHEP

### DISCUSSION THREAD

**Discussion by samardzicneo** [bluenviron/mediamtx#2892] [Questions]
### Question

I'm wondering if it's possible to use MediaMTX' WebRTC streams on a client without WHEP - the reason for this is that we need to support client devies running ancient Chromium versions which have WebRTC but no support for `addTransciever`, which your WHEP implementation seems to rely on.

Is there any info on this?

**Comment by aler9** [comment]
Hello, WHEP doesn't have anything to do with `addTransceiver()`, the two things are unrelated.

The legacy way of initializing a WebRTC Session before the introduction of `addTransceiver()` was to call `createOffer()` with two parameters `offerToReceiveVideo` and `offerToReceiveAudio` , but those have been marked as deprecated, therefore we cannot use them anymore since they could be removed at any time.

My advice is to create a custom web page that makes use of `offerToReceiveVideo` and `offerToReceiveAudio` and use it to read a WebRTC stream from the server.

[a comment by an automated account is not shown]
